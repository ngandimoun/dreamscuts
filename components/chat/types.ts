export interface Message {
  id: string;
  type: "user" | "assistant";
  content: string;
  timestamp: Date;
  mediaUrl?: string;
  isConceptionSummary?: boolean;
  isAssistantResponse?: boolean;
}

export interface GenerationStep {
  id: string;
  title: string;
  description: string;
  status: "pending" | "active" | "completed";
}

export interface ChatState {
  messages: Message[];
  isGenerating: boolean;
  generationSteps: string[];
  currentStep: number;
  finalResult: string | null;
}
