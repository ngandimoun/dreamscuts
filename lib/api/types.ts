// Types pour les APIs de m√©dias

export interface MediaItem {
  id: string;
  name: string;
  type: 'image' | 'video' | 'audio';
  url: string;
  thumbnail: string;
  uploadedAt: Date;
  isGenerated?: boolean;
  source?: 'pexels' | 'pixabay' | 'unsplash' | 'freesound';
  originalData?: any;
}

// Types Pexels
export interface PexelsPhoto {
  id: number;
  width: number;
  height: number;
  url: string;
  photographer: string;
  photographer_url: string;
  photographer_id: number;
  avg_color: string;
  src: {
    original: string;
    large2x: string;
    large: string;
    medium: string;
    small: string;
    portrait: string;
    landscape: string;
    tiny: string;
  };
  liked: boolean;
  alt: string;
}

export interface PexelsVideo {
  id: number;
  width: number;
  height: number;
  url: string;
  image: string;
  full_res: string | null;
  tags: string[];
  duration: number;
  user: {
    id: number;
    name: string;
    url: string;
  };
  video_files: Array<{
    id: number;
    quality: string;
    file_type: string;
    width: number;
    height: number;
    fps: number;
    link: string;
  }>;
  video_pictures: Array<{
    id: number;
    picture: string;
    nr: number;
  }>;
}

export interface PexelsResponse<T> {
  page: number;
  per_page: number;
  photos?: T[];
  videos?: T[];
  total_results: number;
  next_page: string;
}

// Types Pixabay
export interface PixabayImage {
  id: number;
  pageURL: string;
  type: string;
  tags: string;
  previewURL: string;
  previewWidth: number;
  previewHeight: number;
  webformatURL: string;
  webformatWidth: number;
  webformatHeight: number;
  largeImageURL: string;
  fullHDURL: string;
  imageURL: string;
  imageWidth: number;
  imageHeight: number;
  imageSize: number;
  views: number;
  downloads: number;
  likes: number;
  comments: number;
  user_id: number;
  user: string;
  userImageURL: string;
}

export interface PixabayVideo {
  id: number;
  pageURL: string;
  type: string;
  tags: string;
  duration: number;
  videos: {
    large: {
      url: string;
      width: number;
      height: number;
      size: number;
      thumbnail: string;
    };
    medium: {
      url: string;
      width: number;
      height: number;
      size: number;
      thumbnail: string;
    };
    small: {
      url: string;
      width: number;
      height: number;
      size: number;
      thumbnail: string;
    };
    tiny: {
      url: string;
      width: number;
      height: number;
      size: number;
      thumbnail: string;
    };
  };
  views: number;
  downloads: number;
  likes: number;
  comments: number;
  user_id: number;
  user: string;
  userImageURL: string;
}

export interface PixabayResponse<T> {
  total: number;
  totalHits: number;
  hits: T[];
}

// Types Unsplash
export interface UnsplashPhoto {
  id: string;
  created_at: string;
  updated_at: string;
  promoted_at: string | null;
  width: number;
  height: number;
  color: string;
  blur_hash: string;
  description: string | null;
  alt_description: string | null;
  urls: {
    raw: string;
    full: string;
    regular: string;
    small: string;
    thumb: string;
  };
  links: {
    self: string;
    html: string;
    download: string;
    download_location: string;
  };
  user: {
    id: string;
    updated_at: string;
    username: string;
    name: string;
    first_name: string;
    last_name: string | null;
    twitter_username: string | null;
    portfolio_url: string | null;
    bio: string | null;
    location: string | null;
    links: {
      self: string;
      html: string;
      photos: string;
      likes: string;
      portfolio: string;
      following: string;
      followers: string;
    };
    profile_image: {
      small: string;
      medium: string;
      large: string;
    };
    instagram_username: string | null;
    total_collections: number;
    total_likes: number;
    total_photos: number;
    accepted_tos: boolean;
    for_hire: boolean;
    social: {
      instagram_username: string | null;
      portfolio_url: string | null;
      twitter_username: string | null;
      paypal_email: string | null;
    };
  };
  tags: Array<{
    type: string;
    title: string;
  }>;
}

// Types Freesound
export interface FreesoundSound {
  id: number;
  url: string;
  name: string;
  tags: string[];
  description: string;
  type: string;
  duration: number;
  filesize: number;
  bitrate: number;
  bitdepth: number;
  samplerate: number;
  channels: number;
  license: string;
  username: string;
  pack: string | null;
  pack_name: string | null;
  download: string;
  bookmark: string;
  previews: {
    'preview-hq-mp3': string;
    'preview-hq-ogg': string;
    'preview-lq-mp3': string;
    'preview-lq-ogg': string;
  };
  images: {
    waveform_m: string;
    waveform_l: string;
    spectral_m: string;
    spectral_l: string;
  };
  num_downloads: number;
  avg_rating: number;
  num_ratings: number;
  created: string;
  modified: string;
  num_comments: number;
  comment: string;
  original_filename: string;
  filesize_approx: number;
}

export interface FreesoundResponse {
  count: number;
  next: string | null;
  previous: string | null;
  results: FreesoundSound[];
}

