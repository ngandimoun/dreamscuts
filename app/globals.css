@import "tailwindcss";
@import "tw-animate-css";

@custom-variant dark (&:is(.dark *));

:root {
  /* Updated color tokens to match the pastel design brief */
  --background: #ffffff;
  --foreground: #374151;
  --card: #f0fdf4;
  --card-foreground: #374151;
  --popover: #ffffff;
  --popover-foreground: #374151;
  --primary: #d1fae5;
  --primary-foreground: #374151;
  --secondary: #fbcfe8;
  --secondary-foreground: #374151;
  --muted: #f0fdf4;
  --muted-foreground: #6b7280;
  --accent: #fbcfe8;
  --accent-foreground: #374151;
  --destructive: #dc2626;
  --destructive-foreground: #ffffff;
  --border: #e5e7eb;
  --input: #f0fdf4;
  --ring: #d1fae5;
  --chart-1: #d1fae5;
  --chart-2: #fbcfe8;
  --chart-3: #f0fdf4;
  --chart-4: #ffffff;
  --chart-5: #374151;
  --radius: 0.75rem;
  --sidebar: #ffffff;
  --sidebar-foreground: #374151;
  --sidebar-primary: #d1fae5;
  --sidebar-primary-foreground: #374151;
  --sidebar-accent: #fbcfe8;
  --sidebar-accent-foreground: #374151;
  --sidebar-border: #e5e7eb;
  --sidebar-ring: #d1fae5;

  /* Added custom gradient colors for the background */
  --gradient-start: #e0f7fa;
  --gradient-end: #f3e5f5;
  --lavender: #e1bee7;
  --light-teal: #b2dfdb;
}

.dark {
  --background: #010102;
  --foreground: #f1f5f9;
  --card: #1e293b;
  --card-foreground: #f1f5f9;
  --popover: #1e293b;
  --popover-foreground: #f1f5f9;
  --primary: #22c55e;
  --primary-foreground: #0f172a;
  --secondary: #ec4899;
  --secondary-foreground: #f1f5f9;
  --muted: #334155;
  --muted-foreground: #94a3b8;
  --accent: #ec4899;
  --accent-foreground: #f1f5f9;
  --destructive: #ef4444;
  --destructive-foreground: #f1f5f9;
  --border: #334155;
  --input: #334155;
  --ring: #22c55e;
  --chart-1: #22c55e;
  --chart-2: #ec4899;
  --chart-3: #334155;
  --chart-4: #1e293b;
  --chart-5: #f1f5f9;
  --sidebar: #0f172a;
  --sidebar-foreground: #f1f5f9;
  --sidebar-primary: #22c55e;
  --sidebar-primary-foreground: #0f172a;
  --sidebar-accent: #334155;
  --sidebar-accent-foreground: #f1f5f9;
  --sidebar-border: #334155;
  --sidebar-ring: #22c55e;
}

@theme inline {
  --font-sans: var(--font-geist-sans);
  --font-mono: var(--font-geist-mono);
  --color-background: var(--background);
  --color-foreground: var(--foreground);
  --color-card: var(--card);
  --color-card-foreground: var(--card-foreground);
  --color-popover: var(--popover);
  --color-popover-foreground: var(--popover-foreground);
  --color-primary: var(--primary);
  --color-primary-foreground: var(--primary-foreground);
  --color-secondary: var(--secondary);
  --color-secondary-foreground: var(--secondary-foreground);
  --color-muted: var(--muted);
  --color-muted-foreground: var(--muted-foreground);
  --color-accent: var(--accent);
  --color-accent-foreground: var(--accent-foreground);
  --color-destructive: var(--destructive);
  --color-destructive-foreground: var(--destructive-foreground);
  --color-border: var(--border);
  --color-input: var(--input);
  --color-ring: var(--ring);
  --color-chart-1: var(--chart-1);
  --color-chart-2: var(--chart-2);
  --color-chart-3: var(--chart-3);
  --color-chart-4: var(--chart-4);
  --color-chart-5: var(--chart-5);
  --radius-sm: calc(var(--radius) - 4px);
  --radius-md: calc(var(--radius) - 2px);
  --radius-lg: var(--radius);
  --radius-xl: calc(var(--radius) + 4px);
  --color-sidebar: var(--sidebar);
  --color-sidebar-foreground: var(--sidebar-foreground);
  --color-sidebar-primary: var(--sidebar-primary);
  --color-sidebar-primary-foreground: var(--sidebar-primary-foreground);
  --color-sidebar-accent: var(--sidebar-accent);
  --color-sidebar-accent-foreground: var(--sidebar-accent-foreground);
  --color-sidebar-border: var(--sidebar-border);
  --color-sidebar-ring: var(--sidebar-ring);
}

@layer components {
  .chip-button {
    @apply flex items-center bg-white/70 hover:bg-white border border-gray-200/80 rounded-lg px-3 py-1.5 text-sm font-medium text-gray-700 shadow-sm transition-colors;
  }
}

@layer base {
  * {
    @apply border-border outline-ring/50;
  }
  body {
    @apply bg-background text-foreground;
  }
}


.libertinus {
  font-family: "Libertinus Serif Display", system-ui;
  font-weight: 500;
  font-style: normal;
}


.edu {
  font-family: "Edu NSW ACT Cursive", cursive;
  font-optical-sizing: auto;
  font-weight: 400;
  font-style: normal;
}


.mozilla {
  font-family: "Mozilla Headline", sans-serif;
  font-optical-sizing: auto;
  font-weight: 400;
  font-style: normal;
  font-variation-settings:
    "wdth" 100;
}


.ibarra-real-nova {
  font-family: "Ibarra Real Nova", serif;
  font-optical-sizing: auto;
  font-weight: 500;
  font-style: normal;
}

/* Scrollbars personnalisées pour l'interface de chat */
.scrollbar-thin {
  scrollbar-width: thin;
  scrollbar-color: #d1d5db #f3f4f6;
}

.scrollbar-thin::-webkit-scrollbar {
  width: 6px;
}

.scrollbar-thin::-webkit-scrollbar-track {
  background: #f3f4f6;
  border-radius: 3px;
}

.scrollbar-thin::-webkit-scrollbar-thumb {
  background: #d1d5db;
  border-radius: 3px;
}

.scrollbar-thin::-webkit-scrollbar-thumb:hover {
  background: #9ca3af;
}

/* Empêcher le scroll global quand l'interface de chat est active */
body.chat-active {
  overflow: hidden;
}

/* Scroll personnalisé pour les composants */
.custom-scrollbar {
  scrollbar-width: thin;
  scrollbar-color: rgba(156, 163, 175, 0.5) transparent;
}

.custom-scrollbar::-webkit-scrollbar {
  width: 8px;
}

.custom-scrollbar::-webkit-scrollbar-track {
  background: transparent;
}

.custom-scrollbar::-webkit-scrollbar-thumb {
  background: rgba(156, 163, 175, 0.5);
  border-radius: 4px;
}

.custom-scrollbar::-webkit-scrollbar-thumb:hover {
  background: rgba(156, 163, 175, 0.7);
}

/* Désactiver le scroll global pour toute l'application */
body {
  overflow: hidden;
}

/* Styles pour le redimensionnement des colonnes */
.resize-handle {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 4px;
  background: #e5e7eb;
  cursor: col-resize;
  transition: background-color 0.2s ease;
  z-index: 10;
}

.resize-handle:hover {
  background: #3b82f6;
}

.resize-handle:active {
  background: #2563eb;
}

.resize-handle::after {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 2px;
  height: 32px;
  background: #9ca3af;
  border-radius: 1px;
  opacity: 0;
  transition: opacity 0.2s ease;
}

.resize-handle:hover::after {
  opacity: 1;
}

/* Empêcher la sélection de texte pendant le redimensionnement */
body.resizing {
  cursor: col-resize !important;
  user-select: none !important;
}

body.resizing * {
  cursor: col-resize !important;
  user-select: none !important;
}