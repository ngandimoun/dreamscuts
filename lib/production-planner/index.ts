/**
 * ðŸŽ¬ Production Planner - Phase 0 Index
 * 
 * Main entry point for the Production Planner system.
 * Exports all schemas, types, and database services.
 */

// ========================================
// SCHEMAS & VALIDATION
// ========================================
export * from './schemas';

// ========================================
// TYPES
// ========================================
export * from './types';

// ========================================
// DATABASE SERVICES
// ========================================
export * from './database';

// ========================================
// UTILITY FUNCTIONS
// ========================================

/**
 * Create a Production Planner database instance with environment variables
 */
export function createProductionPlannerDatabase() {
  const { createProductionPlannerDatabase: createDB } = require('./database');
  return createDB();
}

/**
 * Validate a production manifest
 */
export function validateManifest(manifestData: unknown) {
  const { validateProductionManifest } = require('./schemas');
  return validateProductionManifest(manifestData);
}

/**
 * Get default production planner configuration
 */
export function getDefaultConfig() {
  return {
    max_concurrent_jobs: 3,
    default_timeout_seconds: 300,
    default_retry_attempts: 3,
    quality_threshold: 0.8,
    strict_validation: true,
    auto_repair_attempts: 2,
    enable_metrics: true,
    log_level: 'info' as const
  };
}

// ========================================
// CONSTANTS
// ========================================

export const PRODUCTION_PLANNER_VERSION = '1.0.0';

export const SUPPORTED_PLATFORMS = [
  'youtube',
  'instagram',
  'tiktok',
  'facebook',
  'twitter',
  'linkedin',
  'web'
] as const;

export const SUPPORTED_ASPECT_RATIOS = [
  '16:9',
  '9:16',
  '1:1',
  '4:3',
  '3:4',
  '21:9'
] as const;

export const SUPPORTED_LANGUAGES = [
  'en',
  'es',
  'fr',
  'de',
  'it',
  'pt',
  'ru',
  'ja',
  'ko',
  'zh',
  'ar',
  'hi'
] as const;

export const DEFAULT_JOB_TIMEOUTS = {
  'voiceover_generation': 120,
  'music_generation': 180,
  'image_generation': 60,
  'video_generation_ai': 300,
  'chart_generation': 30,
  'subtitle_generation': 45,
  'lipsync_processing': 240,
  'visual_effects': 90,
  'asset_enhancement': 60,
  'consistency_check': 30,
  'quality_validation': 45,
  'final_assembly': 180,
  'rendering': 600
} as const;

export const QUALITY_THRESHOLDS = {
  'excellent': 0.9,
  'good': 0.8,
  'acceptable': 0.7,
  'poor': 0.6,
  'unacceptable': 0.5
} as const;
