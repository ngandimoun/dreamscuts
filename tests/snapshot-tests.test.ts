/**
 * Snapshot Tests for Profile-Pipeline Integration
 * 
 * These tests ensure that the profile-pipeline integration produces
 * consistent, deterministic output that can be validated against
 * known good snapshots.
 */

import { describe, test, expect } from '@jest/globals';
import { ProductionManifest, ProfileContext, HardConstraints } from '../types/production-manifest';
import { mergeProfileIntoManifest } from '../services/phase4/manifestBuilder';
import { resolveManifestConflicts } from '../lib/production-planner/conflict-resolution';
import { getFeatureFlagsForProfile } from '../lib/production-planner/feature-flags';

// Test data for snapshot tests
const snapshotTestManifest: ProductionManifest = {
  metadata: {
    manifestId: 'snapshot-test-123',
    version: '1.0.0',
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    duration: 30,
    language: 'en',
    aspectRatio: '16:9',
    platform: 'youtube',
    title: 'Snapshot Test Video',
    description: 'Test video for snapshot validation'
  },
  scenes: [
    {
      id: 'scene_1',
      title: 'Test Scene',
      duration: 30,
      content: 'This is a test scene for snapshot validation.',
      visualElements: [],
      audioElements: [],
      effects: []
    }
  ],
  jobs: [],
  assets: {
    images: [],
    audio: [],
    video: []
  },
  visuals: {
    colorPalette: ['#FF0000', '#00FF00', '#0000FF', '#FFFF00', '#FF00FF'],
    fonts: {
      primary: 'Arial',
      secondary: 'Helvetica'
    },
    style: 'modern'
  },
  effects: {
    allowed: ['fade', 'slide', 'zoom', 'dramatic', 'cinematic_zoom'],
    forbidden: []
  },
  audio: {
    ttsDefaults: {
      voiceId: 'professional_male',
      style: 'dramatic'
    },
    musicDefaults: {
      genre: 'upbeat',
      intensity: 0.8
    }
  }
};

describe('Snapshot Tests - Profile-Pipeline Integration', () => {
  describe('Educational Explainer Profile Snapshot', () => {
    test('should produce consistent educational profile output', async () => {
      const profileId = 'educational_explainer';
      const result = await mergeProfileIntoManifest(snapshotTestManifest, profileId, null);
      
      // Create snapshot of key properties
      const snapshot = {
        metadata: {
          profileId: result.metadata.profileId,
          profileVersion: result.metadata.profileVersion,
          enforcementMode: result.metadata.enforcementMode,
          hardConstraints: result.metadata.hardConstraints,
          profileContext: {
            profileId: result.metadata.profileContext?.profileId,
            profileName: result.metadata.profileContext?.profileName,
            coreConcept: result.metadata.profileContext?.coreConcept,
            visualApproach: result.metadata.profileContext?.visualApproach,
            styleDirection: result.metadata.profileContext?.styleDirection,
            moodGuidance: result.metadata.profileContext?.moodGuidance,
            enhancementPolicy: result.metadata.profileContext?.enhancementPolicy,
            pipelineConfiguration: result.metadata.profileContext?.pipelineConfiguration
          }
        },
        visuals: {
          colorPalette: result.visuals.colorPalette,
          fonts: result.visuals.fonts,
          style: result.visuals.style
        },
        effects: {
          allowed: result.effects.allowed,
          forbidden: result.effects.forbidden
        },
        audio: {
          ttsDefaults: result.audio.ttsDefaults,
          musicDefaults: result.audio.musicDefaults
        }
      };
      
      expect(snapshot).toMatchSnapshot();
    });

    test('should produce consistent educational feature flags', () => {
      const profileId = 'educational_explainer';
      const flags = getFeatureFlagsForProfile(profileId);
      
      const snapshot = {
        promptEnhancementMode: flags.promptEnhancementMode,
        enableWorkerEnhancements: flags.enableWorkerEnhancements,
        enableProfileOverrides: flags.enableProfileOverrides,
        enableCostCaps: flags.enableCostCaps,
        maxCostPerJob: flags.maxCostPerJob,
        maxTotalCost: flags.maxTotalCost,
        enableTimeoutCaps: flags.enableTimeoutCaps,
        maxJobTimeout: flags.maxJobTimeout,
        maxTotalTimeout: flags.maxTotalTimeout,
        enableQualityGates: flags.enableQualityGates,
        minQualityScore: flags.minQualityScore,
        maxRetries: flags.maxRetries
      };
      
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('UGC Testimonial Profile Snapshot', () => {
    test('should produce consistent UGC profile output', async () => {
      const profileId = 'ugc_testimonial';
      const result = await mergeProfileIntoManifest(snapshotTestManifest, profileId, null);
      
      // Create snapshot of key properties
      const snapshot = {
        metadata: {
          profileId: result.metadata.profileId,
          profileVersion: result.metadata.profileVersion,
          enforcementMode: result.metadata.enforcementMode,
          hardConstraints: result.metadata.hardConstraints,
          profileContext: {
            profileId: result.metadata.profileContext?.profileId,
            profileName: result.metadata.profileContext?.profileName,
            coreConcept: result.metadata.profileContext?.coreConcept,
            visualApproach: result.metadata.profileContext?.visualApproach,
            styleDirection: result.metadata.profileContext?.styleDirection,
            moodGuidance: result.metadata.profileContext?.moodGuidance,
            enhancementPolicy: result.metadata.profileContext?.enhancementPolicy,
            pipelineConfiguration: result.metadata.profileContext?.pipelineConfiguration
          }
        },
        visuals: {
          colorPalette: result.visuals.colorPalette,
          fonts: result.visuals.fonts,
          style: result.visuals.style
        },
        effects: {
          allowed: result.effects.allowed,
          forbidden: result.effects.forbidden
        },
        audio: {
          ttsDefaults: result.audio.ttsDefaults,
          musicDefaults: result.audio.musicDefaults
        }
      };
      
      expect(snapshot).toMatchSnapshot();
    });

    test('should produce consistent UGC feature flags', () => {
      const profileId = 'ugc_testimonial';
      const flags = getFeatureFlagsForProfile(profileId);
      
      const snapshot = {
        promptEnhancementMode: flags.promptEnhancementMode,
        enableWorkerEnhancements: flags.enableWorkerEnhancements,
        enableProfileOverrides: flags.enableProfileOverrides,
        enableCostCaps: flags.enableCostCaps,
        maxCostPerJob: flags.maxCostPerJob,
        maxTotalCost: flags.maxTotalCost,
        enableTimeoutCaps: flags.enableTimeoutCaps,
        maxJobTimeout: flags.maxJobTimeout,
        maxTotalTimeout: flags.maxTotalTimeout,
        enableQualityGates: flags.enableQualityGates,
        minQualityScore: flags.minQualityScore,
        maxRetries: flags.maxRetries
      };
      
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Marketing Dynamic Profile Snapshot', () => {
    test('should produce consistent marketing profile output', async () => {
      const profileId = 'marketing_dynamic';
      const result = await mergeProfileIntoManifest(snapshotTestManifest, profileId, null);
      
      // Create snapshot of key properties
      const snapshot = {
        metadata: {
          profileId: result.metadata.profileId,
          profileVersion: result.metadata.profileVersion,
          enforcementMode: result.metadata.enforcementMode,
          hardConstraints: result.metadata.hardConstraints,
          profileContext: {
            profileId: result.metadata.profileContext?.profileId,
            profileName: result.metadata.profileContext?.profileName,
            coreConcept: result.metadata.profileContext?.coreConcept,
            visualApproach: result.metadata.profileContext?.visualApproach,
            styleDirection: result.metadata.profileContext?.styleDirection,
            moodGuidance: result.metadata.profileContext?.moodGuidance,
            enhancementPolicy: result.metadata.profileContext?.enhancementPolicy,
            pipelineConfiguration: result.metadata.profileContext?.pipelineConfiguration
          }
        },
        visuals: {
          colorPalette: result.visuals.colorPalette,
          fonts: result.visuals.fonts,
          style: result.visuals.style
        },
        effects: {
          allowed: result.effects.allowed,
          forbidden: result.effects.forbidden
        },
        audio: {
          ttsDefaults: result.audio.ttsDefaults,
          musicDefaults: result.audio.musicDefaults
        }
      };
      
      expect(snapshot).toMatchSnapshot();
    });

    test('should produce consistent marketing feature flags', () => {
      const profileId = 'marketing_dynamic';
      const flags = getFeatureFlagsForProfile(profileId);
      
      const snapshot = {
        promptEnhancementMode: flags.promptEnhancementMode,
        enableWorkerEnhancements: flags.enableWorkerEnhancements,
        enableProfileOverrides: flags.enableProfileOverrides,
        enableCostCaps: flags.enableCostCaps,
        maxCostPerJob: flags.maxCostPerJob,
        maxTotalCost: flags.maxTotalCost,
        enableTimeoutCaps: flags.enableTimeoutCaps,
        maxJobTimeout: flags.maxJobTimeout,
        maxTotalTimeout: flags.maxTotalTimeout,
        enableQualityGates: flags.enableQualityGates,
        minQualityScore: flags.minQualityScore,
        maxRetries: flags.maxRetries
      };
      
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('UGC Reaction Profile Snapshot', () => {
    test('should produce consistent UGC reaction profile output', async () => {
      const profileId = 'ugc_reaction';
      const result = await mergeProfileIntoManifest(snapshotTestManifest, profileId, null);
      
      // Create snapshot of key properties
      const snapshot = {
        metadata: {
          profileId: result.metadata.profileId,
          profileVersion: result.metadata.profileVersion,
          enforcementMode: result.metadata.enforcementMode,
          hardConstraints: result.metadata.hardConstraints,
          profileContext: {
            profileId: result.metadata.profileContext?.profileId,
            profileName: result.metadata.profileContext?.profileName,
            coreConcept: result.metadata.profileContext?.coreConcept,
            visualApproach: result.metadata.profileContext?.visualApproach,
            styleDirection: result.metadata.profileContext?.styleDirection,
            moodGuidance: result.metadata.profileContext?.moodGuidance,
            enhancementPolicy: result.metadata.profileContext?.enhancementPolicy,
            pipelineConfiguration: result.metadata.profileContext?.pipelineConfiguration
          }
        },
        visuals: {
          colorPalette: result.visuals.colorPalette,
          fonts: result.visuals.fonts,
          style: result.visuals.style
        },
        effects: {
          allowed: result.effects.allowed,
          forbidden: result.effects.forbidden
        },
        audio: {
          ttsDefaults: result.audio.ttsDefaults,
          musicDefaults: result.audio.musicDefaults
        }
      };
      
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Cinematic Trailer Profile Snapshot', () => {
    test('should produce consistent cinematic trailer profile output', async () => {
      const profileId = 'cinematic_trailer';
      const result = await mergeProfileIntoManifest(snapshotTestManifest, profileId, null);
      
      // Create snapshot of key properties
      const snapshot = {
        metadata: {
          profileId: result.metadata.profileId,
          profileVersion: result.metadata.profileVersion,
          enforcementMode: result.metadata.enforcementMode,
          hardConstraints: result.metadata.hardConstraints,
          profileContext: {
            profileId: result.metadata.profileContext?.profileId,
            profileName: result.metadata.profileContext?.profileName,
            coreConcept: result.metadata.profileContext?.coreConcept,
            visualApproach: result.metadata.profileContext?.visualApproach,
            styleDirection: result.metadata.profileContext?.styleDirection,
            moodGuidance: result.metadata.profileContext?.moodGuidance,
            enhancementPolicy: result.metadata.profileContext?.enhancementPolicy,
            pipelineConfiguration: result.metadata.profileContext?.pipelineConfiguration
          }
        },
        visuals: {
          colorPalette: result.visuals.colorPalette,
          fonts: result.visuals.fonts,
          style: result.visuals.style
        },
        effects: {
          allowed: result.effects.allowed,
          forbidden: result.effects.forbidden
        },
        audio: {
          ttsDefaults: result.audio.ttsDefaults,
          musicDefaults: result.audio.musicDefaults
        }
      };
      
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Conflict Resolution Snapshot', () => {
    test('should produce consistent conflict resolution output', () => {
      const profileContext: ProfileContext = {
        profileId: 'educational_explainer',
        profileName: 'Educational Explainer',
        coreConcept: 'Clear, educational content',
        visualApproach: 'Minimalist, clean design',
        styleDirection: 'Professional, accessible',
        moodGuidance: 'Calm, informative',
        pipelineConfiguration: {
          imageModel: 'nano_banana',
          videoModel: 'veo3_fast',
          ttsModel: 'elevenlabs_dialogue'
        },
        enhancementPolicy: 'additive'
      };

      const hardConstraints: HardConstraints = {
        style: {
          palette: ['#FFFFFF', '#000000', '#CCCCCC'],
          visualStyle: 'minimalist'
        },
        effects: {
          forbiddenTypes: ['dramatic', 'cinematic_zoom']
        },
        audioStyle: {
          tone: 'professional',
          voiceStyle: 'clear'
        }
      };

      const result = resolveManifestConflicts(
        snapshotTestManifest,
        profileContext,
        hardConstraints,
        'strict'
      );

      const snapshot = {
        warnings: result.warnings,
        resolvedManifest: {
          visuals: {
            colorPalette: result.resolvedManifest.visuals.colorPalette,
            fonts: result.resolvedManifest.visuals.fonts,
            style: result.resolvedManifest.visuals.style
          },
          effects: {
            allowed: result.resolvedManifest.effects.allowed,
            forbidden: result.resolvedManifest.effects.forbidden
          },
          audio: {
            ttsDefaults: result.resolvedManifest.audio.ttsDefaults,
            musicDefaults: result.resolvedManifest.audio.musicDefaults
          }
        }
      };

      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('UI Override Snapshot', () => {
    test('should produce consistent UI override output', async () => {
      const profileId = 'educational_explainer';
      const uiInput = {
        profileId: 'marketing_dynamic',
        enforcementMode: 'creative' as const
      };
      
      const result = await mergeProfileIntoManifest(snapshotTestManifest, profileId, uiInput);
      
      // Create snapshot of key properties
      const snapshot = {
        metadata: {
          profileId: result.metadata.profileId,
          profileVersion: result.metadata.profileVersion,
          enforcementMode: result.metadata.enforcementMode,
          hardConstraints: result.metadata.hardConstraints,
          profileContext: {
            profileId: result.metadata.profileContext?.profileId,
            profileName: result.metadata.profileContext?.profileName,
            coreConcept: result.metadata.profileContext?.coreConcept,
            visualApproach: result.metadata.profileContext?.visualApproach,
            styleDirection: result.metadata.profileContext?.styleDirection,
            moodGuidance: result.metadata.profileContext?.moodGuidance,
            enhancementPolicy: result.metadata.profileContext?.enhancementPolicy,
            pipelineConfiguration: result.metadata.profileContext?.pipelineConfiguration
          }
        },
        visuals: {
          colorPalette: result.visuals.colorPalette,
          fonts: result.visuals.fonts,
          style: result.visuals.style
        },
        effects: {
          allowed: result.effects.allowed,
          forbidden: result.effects.forbidden
        },
        audio: {
          ttsDefaults: result.audio.ttsDefaults,
          musicDefaults: result.audio.musicDefaults
        }
      };
      
      expect(snapshot).toMatchSnapshot();
    });
  });
});
