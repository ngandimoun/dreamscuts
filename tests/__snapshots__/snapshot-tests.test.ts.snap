// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests - Profile-Pipeline Integration Educational Explainer Profile Snapshot should produce consistent educational feature flags 1`] = `
{
  "enableCostCaps": true,
  "enableProfileOverrides": true,
  "enableQualityGates": true,
  "enableTimeoutCaps": true,
  "enableWorkerEnhancements": true,
  "maxCostPerJob": 0.5,
  "maxJobTimeout": 600,
  "maxRetries": 3,
  "maxTotalCost": 5,
  "maxTotalTimeout": 3600,
  "minQualityScore": 0.3,
  "promptEnhancementMode": "strict",
}
`;

exports[`Snapshot Tests - Profile-Pipeline Integration Marketing Dynamic Profile Snapshot should produce consistent marketing feature flags 1`] = `
{
  "enableCostCaps": true,
  "enableProfileOverrides": true,
  "enableQualityGates": true,
  "enableTimeoutCaps": true,
  "enableWorkerEnhancements": true,
  "maxCostPerJob": 1,
  "maxJobTimeout": 600,
  "maxRetries": 3,
  "maxTotalCost": 10,
  "maxTotalTimeout": 3600,
  "minQualityScore": 0.3,
  "promptEnhancementMode": "creative",
}
`;

exports[`Snapshot Tests - Profile-Pipeline Integration UGC Testimonial Profile Snapshot should produce consistent UGC feature flags 1`] = `
{
  "enableCostCaps": true,
  "enableProfileOverrides": true,
  "enableQualityGates": true,
  "enableTimeoutCaps": true,
  "enableWorkerEnhancements": true,
  "maxCostPerJob": 0.3,
  "maxJobTimeout": 600,
  "maxRetries": 3,
  "maxTotalCost": 3,
  "maxTotalTimeout": 3600,
  "minQualityScore": 0.3,
  "promptEnhancementMode": "balanced",
}
`;
